File: src/Ebay/Server.Application.New/Models/MeasurementInfoWithData.cs
using Server.Domain.Measurements;

namespace Server.Application.New.Models;

public record MeasurementInfoWithData(
    string Id,
    Guid ProductId,
    string? MatchId,
    string? LotId,
    string? Location,
    MeasurementState MeasurementState,
    ProductState ProductState,
    string ManufactureCode,
    DateTime? LastTimeWatchedOnEbay,
    DateTime CreatedAt,
    byte[] Data) :
    MeasurementInfo(
        Id: Id,
        ProductId:
        ProductId,
        MatchId: MatchId,
        LotId: LotId,
        Location: Location,
        MeasurementState: MeasurementState,
        ProductState: ProductState,
        ManufactureCode: ManufactureCode,
        CreatedAt: CreatedAt,
        LastTimeWatchedOnEbay: LastTimeWatchedOnEbay);

File: src/Ebay/Server.Application.New/Models/MeasurementInfoWithData.fs
namespace Server.Application.New.Models

open System
open Server.Domain.Measurements

/// <summary>
/// Информация о замере вместе с бинарными данными файла измерения.
/// </summary>
type MeasurementInfoWithData =
    {
        /// <summary>
        /// Идентификатор замера.
        /// </summary>
        Id: string

        /// <summary>
        /// Идентификатор товара, к которому относится замер.
        /// </summary>
        ProductId: Guid

        /// <summary>
        /// Идентификатор пары замеров.
        /// </summary>
        MatchId: string

        /// <summary>
        /// Идентификатор лота eBay.
        /// </summary>
        LotId: string

        /// <summary>
        /// Локация хранения товара или замера.
        /// </summary>
        Location: string

        /// <summary>
        /// Текущее состояние замера.
        /// </summary>
        MeasurementState: MeasurementState

        /// <summary>
        /// Состояние товара.
        /// </summary>
        ProductState: ProductState

        /// <summary>
        /// Код производителя лампы.
        /// </summary>
        ManufactureCode: string

        /// <summary>
        /// Дата и время последнего обнаружения на eBay.
        /// </summary>
        LastTimeWatchedOnEbay: Nullable<DateTime>

        /// <summary>
        /// Дата и время создания замера.
        /// </summary>
        CreatedAt: DateTime

        /// <summary>
        /// Сырые бинарные данные файла измерений.
        /// </summary>
        Data: byte[]
    }

    /// <summary>
    /// Признак, что замер был замечен на eBay в течение последних 7 дней.
    /// </summary>
    member this.IsPublishedOnEbay =
        this.LastTimeWatchedOnEbay.HasValue
        && this.LastTimeWatchedOnEbay.Value > DateTime.UtcNow.AddDays(-7.0)
